### PR TITLE
Fixing dark mode in storybook docs view

### LINKED ISSUE #19671: [Bug]: Storybook docs colors are not updated with Themes
### Describe the bug

In the storybook, When we switch themes, colors are updated for the story but not in the docs. If the color of the text is the default, it doesn't render properly for dark theme. 

https://github.com/MetaMask/metamask-extension/assets/39872794/db848288-9c78-45ec-9ffd-235459305f1a

### Steps to reproduce

1. Go to storybook
2. Check the docs in the Text

### Error messages or log output

_No response_

### Version

^7.0.11

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**
This PR updates the Storybook configuration to enable Dark Mode in the Docs view, which was previously non-functional. It also removes unnecessary files from the codebase for better code maintainability.

## **Related issues**

Fixes: #19671 

## **Manual testing steps**

1. Go to the storybook build of this PR
2. Turn dark mode on and off buy pressing the 🌞 / 🌙 icon in the top toolbar
3. See that dark mode works for docs pages

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/10de57f4-c527-423b-a1b6-8b96f568b895

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/5bded120-70ea-44fa-8791-6b7f2ec352c9

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .storybook/3.COLORS.stories.mdx [hunk 1]
```diff
@@ -18,7 +18,7 @@ We follow a 3 tiered system for color design tokens and css variables.
 <div
   style={{
     textAlign: 'center',
-    backgroundColor: 'var(--color-background-alternative)',
+    backgroundColor: 'var(--brand-colors-white-white000)',
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:21]: Required to make the text work for this image 

Screenshot 2023-11-11 at 7 47 34 AM

File: .storybook/4.SHADOW.stories.mdx [hunk 2]
```diff
@@ -217,7 +217,7 @@ Using both size and color tokens, different shadows can be applied to components
         boxShadow: 'var(--shadow-size-lg) var(--color-shadow-default)',
       }}
     >
-      <span>Modal</span>
+      Modal
     </div>
   </div>
   <div
```
Comment (georgewrmarshall) [inline .storybook/4.SHADOW.stories.mdx:223]: Updates to fix text in dark mode

Screenshot 2023-11-11 at 7 48 35 AM
Screenshot 2023-11-11 at 7 49 03 AM

File: .storybook/manager.js [hunk 3]
```diff
@@ -1,6 +0,0 @@
-import { addons } from '@storybook/addons';
-import MetaMaskStorybookTheme from './metamask-storybook-theme';
-
-addons.setConfig({
-  theme: MetaMaskStorybookTheme,
-});
```
Comment (georgewrmarshall) [inline .storybook/manager.js:6]: No longer required

File: .storybook/metamask-storybook-theme.js [hunk 4]
```diff
@@ -1,12 +1,6 @@
...
-  base: 'light',
+// MetaMask Storybook Theme
+export const metamaskStorybookTheme = {
   brandTitle: 'MetaMask Storybook',
-
   // Typography
   fontBase: 'Euclid Circular B, Roboto, Helvetica, Arial, sans-serif',
-  fontCode: 'Inconsolata, monospace',
-});
+};
```
Comment (georgewrmarshall) [inline .storybook/metamask-storybook-theme.js:6]: Can just be an object now as the theme is handled in preview.js

File: .storybook/preview-head.html [hunk 5]
```diff
@@ -1,27 +0,0 @@
...
-
-  .docblock-source code {
-    font-family: var(--font-family-monospace) !important;
-    font-size: var(--font-size-code) !important;
-  }
-  .docblock-source code * {
-    font-family: var(--font-family-monospace) !important;
-    font-size: var(--font-size-code) !important;
-  }
-</style>
```
Comment (georgewrmarshall) [inline .storybook/preview-head.html:27]: To make the code blocks work for dark mode it's easier to remove and just accept the default styles

### Before

Screenshot 2023-11-11 at 10 17 39 AM

### After

Screenshot 2023-11-11 at 10 17 19 AM
Screenshot 2023-11-11 at 10 18 17 AM

File: .storybook/preview.js [hunk 6]
```diff
@@ -28,7 +30,23 @@ export const parameters = {
...
+      const props = {
+        ...context,
+        theme: isDark
+          ? { ...themes.dark, ...metamaskStorybookTheme }
+          : { ...themes.light, ...metamaskStorybookTheme },
+        'data-theme': isDark ? 'dark' : 'light',
+      };
+
+      return (
+        <div data-theme={isDark ? 'dark' : 'light'}>
```
Comment (georgewrmarshall) [inline .storybook/preview.js:45]: Handles the design tokens theme change

File: .storybook/preview.js [hunk 7]
```diff
@@ -28,7 +30,23 @@ export const parameters = {
...
   docs: {
-    theme: MetaMaskStorybookTheme,
+    container: (context) => {
+      const isDark = useDarkMode();
+
+      const props = {
+        ...context,
+        theme: isDark
+          ? { ...themes.dark, ...metamaskStorybookTheme }
+          : { ...themes.light, ...metamaskStorybookTheme },
```
Comment (georgewrmarshall) [inline .storybook/preview.js:40]: Updates colors as but adheres to brand fonts and some other small theme customizations

File: .storybook/preview.js [hunk 8]
```diff
@@ -81,16 +99,10 @@ const proxiedBackground = new Proxy(
...
+  const isDark = useDarkMode();
   const currentLocale = context.globals.locale;
   const current = allLocales[currentLocale];
 
-  useEffect(() => {
-    channel.on('DARK_MODE', setDark);
-    return () => channel.off('DARK_MODE', setDark);
-  }, [channel, setDark]);
-
   useEffect(() => {
```
Comment (georgewrmarshall) [inline .storybook/preview.js:106]: No longer required for theming to work

**Comment by georgewrmarshall** [comment]
Note: A non-critical bug causing a slight styling disparity on anchor tags at the start of each heading when the theme changes in the initial doc pages exists. The issue self-resolves upon refreshing the page. Despite this minor glitch, IMO this PR should not be blocked due to its overall improvements.

https://github.com/MetaMask/metamask-extension/assets/8112138/d1c119c0-d206-4269-805d-bc5516843bc9

File: .storybook/4.SHADOW.stories.mdx [hunk 9]
```diff
@@ -42,10 +42,10 @@ There are 4 different sizes of shadow in MetaMask
       display: 'grid',
       alignContent: 'center',
       justifyContent: 'center',
-      spanAlign: 'center',
```
Comment (georgewrmarshall) [inline .storybook/4.SHADOW.stories.mdx:45]: removing incorrect styling

File: .storybook/4.SHADOW.stories.mdx [hunk 10]
```diff
@@ -28,10 +28,10 @@ There are 4 different sizes of shadow in MetaMask
...
```
Comment (georgewrmarshall) [inline .storybook/4.SHADOW.stories.mdx:31]: removing incorrect styling

File: package.json [hunk 11]
```diff
@@ -402,6 +402,7 @@
     "@storybook/addon-a11y": "^7.4.6",
     "@storybook/addon-actions": "^7.4.6",
     "@storybook/addon-designs": "^7.0.5",
+    "@storybook/addon-docs": "^7.4.6",
```
Comment (georgewrmarshall) [inline package.json:405]: Fixing dependency issue

Screenshot 2023-11-17 at 3 33 15 PM

**Comment by georgewrmarshall** [comment]
@metamaskbot update-policies

**Review by garrettbear — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**